daemon/daemon.go

```diff
@@ -905,6 +883,51 @@ func (d *Daemon) init() error {
 	return nil
 }
 
+// startLXCMapSync spawns a controller which synchronizes the host entries in BPF lxc Map
+// with the in-memory host IP address values.
+func (d *Daemon) startLXCMapSync() error {
+	// Initially, clean all endpoint entries
+	if err := lxcmap.LXCMap.DeleteAll(); err != nil {
+		return err
+	}
+
+	controller.NewManager().UpdateController("lxcmap-bpf-host-sync",
+		controller.ControllerParams{
+			DoFunc: func() error {
+				// TODO: Update addresses first, in case node addressing has changed.
+				// TODO: Once these start changing on runtime, figure out the locking strategy.
+				// TODO: Delete stale host entries from the lxcmap.
+				localIPs := []net.IP{
+					node.GetInternalIPv4(),
+					node.GetExternalIPv4(),
+					node.GetIPv6(),
+					node.GetIPv6Router(),
+				}
+
+				for _, ip := range localIPs {
+					if err := lxcmap.SyncHostEntry(ip); err != nil {
+						return fmt.Errorf("Unable to add host entry to endpoint map: %s", err)
+					}
+					id, exists := ipcache.IPIdentityCache.LookupByIP(ip.String())
+					if !exists || id != identity.ReservedIdentityHost {
+						// Upsert will not propagate (reserved:host->ID) mappings across the cluster.
+						// Manually push it into each IPIdentityMappingListener.
+						log.WithField(logfields.IPAddr, ip).Debug("Adding local ip to ipcache")
+						ipcache.IPIdentityCache.Upsert(ip.String(), identity.ReservedIdentityHost)
+
+						localIPIDPair := identity.IPIdentityPair{IP: ip, ID: identity.ReservedIdentityHost}
+						d.OnIPIdentityCacheChange(ipcache.Upsert, localIPIDPair)
+						envoy.NetworkPolicyHostsCache.OnIPIdentityCacheChange(ipcache.Upsert, localIPIDPair)
+					}
+				}
+				return nil
+			},
+			RunInterval: time.Duration(5) * time.Second,
+		})
+	// TODO: To keep the old behavior we should wait for one successful completion of the controller
```

Comment:
We can't keep this as a todo as we might get into a situation where no host IP are synced down.

You can do the following:
- extract the controller run function into a separate function and run it once and fail on any error to keep behaviour on failing while bootstrapping when any of the host IPs can't be added.
- or create channel and pass it into the controller and then close it on the first successful run. You could then wait on the channel to be closed outside with a timeout. This would give us retries even on bootstrapping but the logic is a bit more complicated.

Reply:
I'll do the first, as it is less complicated.